apps/main/src/llamalend/features/bands-chart/chartOptions.ts

```diff
@@ -0,0 +1,192 @@
+import type { EChartsOption } from 'echarts-for-react'
+import { formatNumber } from '@ui/utils'
+import { generateMarkLines, createLabelStyle } from './markLines'
+import { ChartDataPoint, BandsChartPalette, DerivedChartData, UserBandsPriceRange } from './types'
+
+/**
+ * Creates series data with conditional styling for liquidation bands
+ */
+const createSeriesData = (data: number[], derived: DerivedChartData, color: string, liquidationOutlineColor: string) =>
+  data.map((value: number, index: number) => ({
+    value,
+    itemStyle: {
+      color,
+      borderColor: derived.isLiquidation[index] ? liquidationOutlineColor : 'transparent',
+      borderWidth: derived.isLiquidation[index] ? 2 : 0,
+    },
+  }))
+
+/**
+ * Creates base series configuration with common properties
+ */
+const createSeriesConfig = (
+  name: string,
+  data: unknown[],
+  markArea?: Record<string, unknown> | null,
+  markLine?: Record<string, unknown> | null,
+) => {
+  const baseConfig = {
+    name,
+    type: 'bar' as const,
+    stack: 'total',
+    animation: false,
+    data,
+    animationDuration: 300,
+  }
+
+  return {
+    ...baseConfig,
+    ...(markArea ? { markArea } : {}),
+    ...(markLine ? { markLine } : {}),
```

Comment:
It's fine to `...undefined`
```suggestion
    ...(markArea && { markArea }),
    ...(markLine && { markLine }),
```